test/c/test_split_2q_unitaries.c

```diff
@@ -0,0 +1,192 @@
+    QkCircuit *qc = qk_circuit_new(2, 0);
+    QkComplex64 c0 = {0., 0.};
+    QkComplex64 neg_im = {0., -1.};
+    QkComplex64 im = {0., 1.};
+    QkComplex64 unitary[16] = {
+        c0, c0, c0, neg_im, c0, c0, neg_im, c0, c0, im, c0, c0, im, c0, c0, c0,
+    };
+    uint32_t qargs[2] = {0, 1};
+    qk_circuit_unitary(qc, unitary, qargs, 2, true);
+    QkTranspileLayout *split_result =
+        qk_transpiler_pass_standalone_split_2q_unitaries(qc, 1 - 1e-16, true);
+    int result = Ok;
+    if (split_result != NULL) {
+        result = EqualityError;
+        printf("Permutation returned for a circuit that shouldn't isn't a swap equivalent\n");
+        goto cleanup;
+    }
+    QkOpCounts counts = qk_circuit_count_ops(qc);
+    if (counts.len != 1) {
+        printf("More than 1 type of gate in the circuit\n");
+        result = EqualityError;
+        goto ops_cleanup;
+    }
+    for (int i = 0; i < counts.len; i++) {
+        int gate = strcmp(counts.data[i].name, "unitary");
+        if (gate != 0) {
+            printf("Gates outside expected set in output circuit\n");
+            result = EqualityError;
+            goto ops_cleanup;
+        }
+        unsigned int count = counts.data[i].count;
+        if (count != 2) {
+            printf("Unexpected gate counts found\n");
+            result = EqualityError;
+            goto ops_cleanup;
+        }
+    }
+    QkCircuitInstruction inst;
+    for (size_t i = 0; i < qk_circuit_num_instructions(qc); i++) {
+        qk_circuit_get_instruction(qc, i, &inst);
+        if (inst.num_qubits != 1) {
+            printf("Gate %d operates on more than 1 qubit: %zu\n", i, inst.num_qubits);
+            result = EqualityError;
+            goto ops_cleanup;
+        }
+        qk_circuit_instruction_clear(&inst);
+    }
+
+ops_cleanup:
+    qk_opcounts_free(counts);
+cleanup:
+    qk_circuit_free(qc);
+    return result;
+}
+
+int test_split_2q_unitaries_swap_x_y_unitary(void) {
+    QkCircuit *qc = qk_circuit_new(2, 0);
+    QkComplex64 c0 = {0., 0.};
+    QkComplex64 neg_im = {0., -1.};
+    QkComplex64 im = {0., 1.};
+    QkComplex64 unitary[16] = {
+        c0, c0, c0, neg_im, c0, neg_im, c0, c0, c0, c0, im, c0, im, c0, c0, c0,
+    };
+    uint32_t qargs[2] = {0, 1};
+    qk_circuit_unitary(qc, unitary, qargs, 2, true);
+    QkTranspileLayout *split_result =
+        qk_transpiler_pass_standalone_split_2q_unitaries(qc, 1 - 1e-16, true);
+    int result = Ok;
+    if (split_result == NULL) {
+        result = EqualityError;
+        printf("Permutation returned for a circuit that shouldn't split\n");
+        goto cleanup;
+    }
+    uint32_t permutation[2];
+    qk_transpile_layout_output_permutation(split_result, permutation);
+    uint32_t expected[2] = {1, 0};
+    for (int i = 0; i < 2; i++) {
+        if (permutation[i] != expected[i]) {
+            printf("Permutation at position %d not as expected, found %zu expected %zu\n", i,
+                   permutation[i], expected[i]);
```

Comment:
```suggestion
            printf("Permutation at position %d not as expected, found %u expected %u\n", i,
                   permutation[i], expected[i]);
```
Compiler screams: _use the right types!_